0020-bluetooth-payment-protocol/0020-bluetooth-payment-protocol.md

```diff
@@ -0,0 +1,110 @@
+# Bluetooth Payment Protocol
+
+This document specifies a protocol for communicating and authorizing Interledger Payment Requests over Bluetooth (Low Energy). It is primarily intended for use in in-person retail payment contexts.
+
+This protocol may be implemented on Customer devices such as smartphones or very low-power offline devices as our example[tapFi](https://github.com/joaopedrovbs/tapFi).
+
+Once a Customer becomes active by broadcasting it's payment service over BLE, Merchants can connect
+and request signature for payments over a secure Bluetooth channel. The channel is closed once
+a payment is signed or the Customer cancels the purshase for security or authentication reasons.
+
+## Definitions
+
+- `IPR`: [Interledger Payment Request](../0011-interledger-payment-request/0011-interledger-payment-request.md)
+- `Merchant`: The one requesting a payment
+- `Customer`: The one authorizing and remitting payment
+- `Connection`: Short time binding between `Merchant` and `Customer`
+- `ILP Account`: An account on a Interledger-enabled ledger
+- `ILP Address`: [Interledger](../0015-ilp-addresses/0015-ilp-addresses.md) address in the form of `g.us.somebank.youraccount`
+- `Merchant Certificate Authority`: Like CA's for TLS certificates, Merchant CAs sign the certificates of trusted merchants. CAs MUST maintain blacklists of compromised keys
+- `Redemption URI`: URI where the Merchant can submit Customer-authorized IPRs to initiate an ILP payment to their account (Note: the service that implements this MUST store previously redeemed IPRs to avoid duplicate payments)
+- `Write Long Characteristic`: GATT Sub-Procedure that Writes to Characteristics with more than 20 bytes
```

Comment:
It is (take a look for an example at page 7 on the PDF about "User Data Service (UDS)")
http://www.cypress.com/documentation/software-and-drivers/bluetooth-low-energy-ble-profiles-and-services